### ISSUE TITLE
iOS - "Wrong JPEG library version: library is 90, caller expects 62"

### ISSUE THREAD

**Issue by ilijapuaca** [mozilla/mozjpeg#238]
This might not be the right place to post this, but I'm banging my head for 2 days straight trying to figure this one out.

I've built mozjpeg for iOS per instructions provided in BUILDING.txt file. Everything went fine and I wrote some code that should perform the encoding. When trying to retrieve the handle using `tjInitCompress`, it returns `NULL` and the error string is the one from the title.

I'm not that knowledgable when it comes to exact details about how building/linking works for this particular library, but as far as I've figured it builds libjpeg from source code that is part of the repo. Does it for whatever reason have dynamic link to system libjpeg that would use the one that's available on iOS?

I understand that there are no plans for turbojpeg to support jpeg9 for somewhat obvious reasons, with that in mind, should it be possible to get mozjpeg to work with iOS10/XCode8 setup?

**Comment by CoolOppo** [comment]
Does libjpeg-turbo work if you try to drop it in as a replacement for mozjpeg? If not, your issue is not localized to mozjpeg.

**Comment by CoolOppo** [comment]
Keep in mind that the BUILDING.txt is sort of outdated and contains a lot of leftover (and perhaps untested) things from libjpeg-turbo, since mozjpeg is just a fork of that.

**Comment by ilijapuaca** [comment]
I understand, and the issue is probably better suited for libjpeg-turbo repo (I've also opened an issue there).

Did you guys try running mozjpeg on iOS though? I assume the error is a fairly obvious one for someone that is familiar with the way library is being built. I'd love to make a contribution that would fix the issue if I would get some guidelines as to what exactly is going on.

**Comment by ilijapuaca** [comment]
I eventually got it working by linking with both libjpeg.a and libjpeg-turbo.a that were the result of the build. I assumed it was only libjpeg-turbo.a that I needed and that's the one I created a .framework file out of.

The issue is obviously not with neither libjpeg-turbo nor mozjpeg. I still don't quite get what libjpeg was being linked against when I was getting the error initially, but I'm closing the issue down.
